### PR TITLE
Change "File Size" column name to "Size" at small widths

### LINKED ISSUE #16734: Rename "File Size" column in file browser to "Size"
## Description

It is possible to read file sizes when the file size column in the filebrowser is very narrow, so much so that the word "Size" isn't visible in the column header:

image

Shorten this column name to "Size" — there is no ambiguity here. 

For comparison, "Size" is used as a column header in macOS's Finder:

image

## Context

- Operating System and version: macOS Sonoma 14.6.1
- Browser and version: Firefox ESR 129.0.1
- JupyterLab version: 4.3.0beta1

### PR DESCRIPTION
**Pull request by JasonWeill**
This is a copy of #16749 by @ChairBorn, with the latter two commits backed out and rebased to upstream `main`. Creating to run the test pipelines independently.

Fixes #16734.
Closes #16749

### CODE REVIEW

**Comment by jtpio** [comment]
Looks like the UI tests failures are relevant, and related to the CSS changes in this PR?

[filebrowser-diff.webm](https://github.com/user-attachments/assets/80a6bdb6-a2ce-454f-b492-ec7d38cc4da1)

**Comment by ChairBorn** [comment]
on it, i will double check the tests results before bothering you next time

File: packages/filebrowser/style/base.css [hunk 1]
```diff
@@ -196,42 +197,38 @@
...
   }
 }
 
-.jp-DirListing-headerItem.jp-id-filesize {
-  text-align: right;
-}
-
-.jp-DirListing-headerItem.jp-mod-selected {
-  font-weight: 600;
-}
```
Comment (krassowski) [inline packages/filebrowser/style/base.css:233]: This should be preserved

**Comment by jtpio** [comment]
Hmm, is this change caught by the examples UI tests intended?

| Actual | Expected | 
| --------- | --------------|
| example-actual | example-expected |

File: packages/filebrowser/style/base.css [hunk 2]
```diff
@@ -196,38 +197,38 @@
  * Use container queries (not yet supported by our version of stylelint)
  * to display either a small or large header for the last-modified column.
  */
-/* stylelint-disable */
```
Comment (jtpio) [inline packages/filebrowser/style/base.css:199]: Looks like we also need to keep this comment:

image

**Comment by jtpio** [comment]
Testing this PR in a clean environment, it looks like there is a glitch where both "Modified" and "Last Modified" appear to be displayed at a specific column width:

[jupyterlab-modified-column-issue.webm](https://github.com/user-attachments/assets/989175b0-da53-4d52-b86b-3482e9b2a4b6)

Which does not seem to be the case with `4.3.0b3`.

**Review by jtpio — APPROVED** [review]
Thanks @ChairBorn and @JasonWeill!

I pushed a couple more fixes and the UI tests are now passing.

**Pull request merged**